src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,276 @@
+        boolean emptyDescription = false;
+        switch (type) {
+        case CHAR_TYPE_TODO:
+            task = new Todo(userParameter);
+            break;
+        case CHAR_TYPE_EVENT:
+            userArguments = userParameter.split(EVENT_STRING_SPLIT_REGEX, 2);
+            regexNotFound = userArguments.length != 2;
+            emptyDescription = userArguments[0].isBlank();
+            if (emptyDescription || regexNotFound) {
+                throw new InvalidParameterException(EVENT_ERROR_MESSAGE);
+            }
+            task = new Event(userArguments[0], userArguments[1]);
+            break;
+        case CHAR_TYPE_DEADLINE:
+            userArguments = userParameter.split(DEADLINE_STRING_SPLIT_REGEX, 2);
+            regexNotFound = userArguments.length != 2;
+            emptyDescription = userArguments[0].isBlank();
+            if (emptyDescription || regexNotFound) {
+                throw new InvalidParameterException(DEADLINE_ERROR_MESSAGE);
+            }
+            task = new Deadline(userArguments[0], userArguments[1]);
+            break;
+        }
+        return task;
+    }
+
+
+    /**
+     * Adds a newly created task with given task description and type to tasks list
+     *
+     * @param userInputArray Command and parameters given by user
+     * @param type           Type of task given by user input
+     */
+    public void addTask(String[] userInputArray, TaskType type) {
+
+        // Check for empty description and return if no description
+        try {
+            boolean emptyDescription = userInputArray[1].isBlank();
+            if (emptyDescription) {
+                Duke.printMessage(EMPTY_DESCRIPTION_ERROR_MESSAGE);
+                return;
+            }
+
+            // Create tasks
+            Task task = null;
+            switch (type) {
+            case TODO:
+                task = createTask(userInputArray[1], CHAR_TYPE_TODO);
+                break;
+            case EVENT:
+                task = createTask(userInputArray[1], CHAR_TYPE_EVENT);
+                break;
+            case DEADLINE:
+                task = createTask(userInputArray[1], CHAR_TYPE_DEADLINE);
+                break;
+            }
+
+            // As long as task is valid, add to task list and inform user
+            if (task != null) {
+                tasksList.add(task);
+                Duke.printMessage(String.format(ADD_TASK_MESSAGE, task, tasksList.size()));
+            }
+
+        } catch (IndexOutOfBoundsException e) {
+            Duke.printMessage(EMPTY_DESCRIPTION_ERROR_MESSAGE);
+        } catch (InvalidParameterException e) {
+            Duke.printMessage(e.getMessage());
+        }
+
+    }
+
+    /**
+     * Prints the entire list of tasks currently in tasks list. Will let user know if there is no task in tasks list.
+     */
+    public void listTask() {
+        int tasksListSize = tasksList.size();
+        String message = "";
+        switch (tasksListSize) {
```

Comment:
Since you are only checking 2 cases, instead of using switch, using `if-else` might be better here.